src/NuGet.Core/NuGet.Common/PathResolver.cs

```diff
@@ -57,6 +57,7 @@ private static Regex WildcardToRegex(string wildcard)
             {
                 // regex wildcard adjustments for *nix-style file systems
                 pattern = pattern
+                    .Replace(@"\.\*\*", @"\.[^/.]*") // .** should not match on ../file or ./file but will match .file
```

Comment:
I'm pretty sure that Windows is already working correctly but I'll reverify tomorrow.